components/nexus-blobstore-file/src/main/java/org/sonatype/nexus/blobstore/file/MapdbBlobMetadataStore.java

```diff
@@ -437,6 +444,23 @@ public long getTotalSize() {
   }
 
   @Override
+  public long getBlobSize() {
+    ensureStarted();
+    return database.execute(new Fun.Function1<Long, DB>()
+    {
+      @Override    
+      public Long run(final DB db) {
+        Long totalSize = 0L;
```

Comment:
Consider using primitive long here to avoid continual auto-boxing/unboxing cost during the loop.